cmd/device-worker/main.go

```diff
@@ -97,14 +99,21 @@ func main() {
 	wl.RegisterObserver(dataMonitor)
 	dataMonitor.Start()
 
-	hbs := heartbeat2.NewHeartbeatService(c, configManager, wl, &hw, dataMonitor)
+	hbs := heartbeat2.NewHeartbeatService(dispatcherClient, configManager, wl, &hw, dataMonitor)
```

Comment:
we need to terminate also dataMonitor that runs its `syncPaths()` on its own goroutine